spec/models/charge_mode_spec.rb

```diff
@@ -0,0 +1,44 @@
+require "rails_helper"
+
+RSpec.describe ChargeMode do
+
+  describe ".for_order_detail" do
+    subject(:charge_for) { described_class.for_order_detail(order_detail) }
+
+    describe "it has an assigned price policy" do
+      let(:order_detail) { build(:order_detail, price_policy: price_policy) }
+
+      describe "for an item" do
```

Comment:
An order detail never has a bundle as its product. If you buy a bundle of say "Item 1" and "Service 2", you end up with two order details, one for each product.